pkgs/development/web/nodejs/nodejs.nix

```diff
@@ -294,18 +294,13 @@ let
         "test-child-process-exec-env"
         "test-child-process-uid-gid"
         "test-fs-write-stream-eagain"
-        "test-https-client-checkServerIdentity"
-        "test-https-foafssl"
-        "test-https-strict"
         "test-process-euid-egid"
         "test-process-initgroups"
         "test-process-setgroups"
         "test-process-uid-gid"
         "test-setproctitle"
-        "test-tls-cli-max-version-1.3"
-        "test-tls-client-auth"
+        # The next test will be fixed by https://github.com/nodejs/node/pull/55089:
```

Comment:
Do we want to just vendor this patch since we’re applying so many anyway? It’ll never be merged into 18.x or 20.x, presumably.

Reply:
> It’ll never be merged into 18.x or 20.x, presumably.

No reason it wouldn't. For v20.x, all (non-semver-minor) patches are typically always backported after a maturation period until it reaches maintenance (next month). For v18.x, I've opened https://github.com/nodejs/node/pull/55101 and plan to include it as well – having passing tests with OpenSSL 3.2 is useful not just for the Nix project.
It will definitely land on v22.x anyway, which is also affected.


> Do we want to just vendor this patch since we’re applying so many anyway? 

I'm not against it, but also I don't think it's necessary, it seems fine to keep the test on that list while we wait for the fix to land upstream. The minimum wait time is 48 hours for this kind of PRs in the Node.js project, so depending on whether this PR has landed, I'll either update it or I'll open a new one targeting `staging` once that happens.

Reply:
Ah, okay; I didn’t realize the LTS branches got so much attention :)

It should be okay to just roll it into this PR once merged. We just threw away all the builds to revert the GCC bump, so it’s going to be quite a few days before eating a Node.js rebuild becomes unpalatable; if you’re okay waiting for the PR to be merged then I’m fine merging this after that.